### ISSUE TITLE
Headless Server Config Schema Documentation

### ISSUE THREAD

**Issue by SnooperXP** [Neos-Metaverse/NeosPublic#2985]
### Where have you searched for this information before making this issue?

I have searched on Neos wiki and found the page on the [Headless Server](https://wiki.neos.com/Headless_Client/Server/en), and on Discord.
I did find some information that looks a bit outdated in `#headless-client`, where it linked to some of the config schema's, as well as some steam commands.

### Which part of the system do you need information on?

I would appreciate more in-depth information on the Headless server configuration json.

### What information do you need to know?

In particular I would like to have:
- A description of each field, what it is and what it does
- The type of the field. ex. boolean, float, string...
- Whether the field is required, or wether the requirement for this field is linked to another. ex. loadWorldUrl and loadWorldPresetName
- Where needed, an example configuration of the field and where to find the information required. ex. loadWorldUrl requiring a world url, and how to retrieve it; or defaultUserRoles and which username that needs to be used

While this does exist for some fields the documentation seems incomplete.

An example of how I think this information could be displayed more effectively is in tables, like the example below:

| Name  | Type | Required | Description | Example |
| :---- | :--- | :------- | :---------- | :------ |
| loadWorldPresetName | Enum | Only if loadWorldURL is undefined(?) | Sets the server to load a new world based on the existing Neos world templates. | The available templates are:<br />-`SpaceWorld`<br />-`Basic Empty`<br />-`GridSpace`<br />-`Microworld`<br />-`Testing Scaling`<br />-`ScratchSpace`<br />-`ScratchSpace (mobile)`<br />-`Instancing Test`<br />-`Physics Locomotion Test`<br />-`UIX Test`
| accessLevel | Enum | Yes | The access level of the session, which sets who can join. | `Private` : Private(Invite Only)<br />`LAN`: LAN Connnections Only<br />`Friends` : Contacts<br />`FriendsOfFriends` : Contacts+<br />`RegisteredUsers`: Registered Users<br />`Anyone` : Anyone |
| defaultUserRoles | Map<String, String> | Yes? | Defines default access levels for users listed when they join the world. | The format is `"USERNAME" : "ROLE"`.<br />Valid roles are:<br />- `Admin`<br />- `Builder`<br />- `Moderator`<br />- `Guest`<br />- `Spectator` |

### Do you plan to format and contribute the provided information to Wiki?

Yes

**Comment by ProbablePrime** [comment]
This page is on my list to update.

However, and generically. Tables are really bad for explaining information of this magnitude. I have some plans for making it nice and clear.

**Comment by Enverex** [comment]
One of the main things that is missing right now is the URL format for loading existing worlds, the key bit being the neosrec:// part (given that if people grab the world URL from Neos in a session, it'll give you the https:// version which won't work).

**Comment by SnooperXP** [comment]
> This page is on my list to update.
> 
> However, and generically. Tables are really bad for explaining information of this magnitude. I have some plans for making it nice and clear.

Fair enough, I only offered up that suggestion as that's what I'm used to seeing on existing developer docs for libraries, though those are usually auto generated.
